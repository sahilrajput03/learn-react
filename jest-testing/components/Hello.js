export const Hello = ({message}) => {
	return <div>Hello {message}</div>
}

export default Hello
