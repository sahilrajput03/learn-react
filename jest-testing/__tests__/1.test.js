// @ts-nocheck
import renderer from 'react-test-renderer'
import Hello from '../components/Hello'

it('renders world message correctly', () => {
	const tree = renderer.create(<Hello message='world' />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('renders sahil message correctly', () => {
	const tree = renderer.create(<Hello message='Sahil' />).toJSON()
	expect(tree).toMatchSnapshot()
})
