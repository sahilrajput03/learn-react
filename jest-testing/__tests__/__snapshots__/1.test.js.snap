// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sahil message correctly 1`] = `
<div>
  Hello 
  Sahil
</div>
`;

exports[`renders world message correctly 1`] = `
<div>
  Hello 
  world
</div>
`;
